rasa_core/training/generator.py

```diff
@@ -365,6 +367,22 @@ def generate(self, silent: bool = False) -> List[TrackerWithCachedStates]:
         logger.debug("Found {} training trackers."
                      "".format(len(finished_trackers)))
 
+        if self.config.augmentation_factor > 0:
+            augm_finished_trackers, original_trackers = [], []
+            for t in finished_trackers:
+                if t.augmented:
+                    augm_finished_trackers.append(t)
+                else:
+                    original_trackers.append(t)
+            augm_finished_trackers = self._subsample_trackers(
+                augm_finished_trackers)
```

Comment:
```suggestion
                augmented_trackers, self.config.max_number_of_augmented_trackers)
```
and in the other places as well,